warp10/src/main/java/io/warp10/script/ext/http/HTTP.java

```diff
@@ -0,0 +1,533 @@
+      conn.setRequestMethod(method.toUpperCase());
+
+      //
+      // Set body
+      //
+
+      if ("GET" != method && "DELETE" != method && "TRACE" != method && "OPTIONS" != method && "HEAD" != method) {
+
+        if (body instanceof String) {
+          String bodyS = (String) body;
+          conn.setDoOutput(bodyS.length() > 0);
+          if (bodyS.length() > 0) {
+            try (OutputStream os = conn.getOutputStream()) {
+              os.write(bodyS.getBytes(StandardCharsets.UTF_8));
+            }
+          }
+
+        } else if (body instanceof byte[]) {
+          byte[] bodyB = (byte[]) body;
+          conn.setDoOutput(bodyB.length > 0);
+          if (bodyB.length > 0) {
+            try (OutputStream os = conn.getOutputStream()) {
+              os.write(bodyB);
+            }
+          }
+
+        } else if (null != body) {
+          throw new WarpScriptException(getName() + " expects the body of the request to be a STRING or BYTES object.");
+        }
+
+      } else if (null != body) {
+        throw new WarpScriptException(getName() + " " + method + " cannot be used with a body.");
+      }
+
+      //
+      // Form response
+      //
+
+      res.put(STATUS_CODE, conn.getResponseCode());
+      Map<String, List<String>> hdrs = conn.getHeaderFields();
+
+      // headers map is immutable
+      res.put(RESPONSE_HEADERS, hdrs);
+
+      // also put status_message in a separate field
+      if (hdrs.containsKey(null)) {
+        List<String> statusMsg = hdrs.get(null);
+        if (statusMsg.size() > 0) {
+          res.put(STATUS_MESSAGE, statusMsg.get(0));
+        } else {
+          res.put(STATUS_MESSAGE, "");
+        }
+      } else {
+        res.put(STATUS_MESSAGE, "");
+      }
+
+      //
+      // Read response
+      //
+
+      InputStream in = null;
+
+      // When there is an error (response code is 404 for instance), body is in the error stream.
+      try {
+        in = conn.getInputStream();
+      } catch (IOException ioe) {
+        in = conn.getErrorStream();
+      }
+
+      if (null == chunkSize) {
+
+        if (null != in) {
+          byte[] buf = new byte[8192];
+          ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+          while (true) {
+            int len = in.read(buf);
+            if (len < 0) {
+              break;
+            }
+
+            if (downloadSize.get() + baos.size() + len > maxsize) {
+              throw new WarpScriptException(getName() + " would exceed maximum size of content which can be retrieved via this function (" + maxsize + " bytes) per script execution.");
+            }
+
+            baos.write(buf, 0, len);
+          }
+
+          downloadSize.addAndGet(baos.size());
+          res.put(CONTENT, baos.toByteArray());
+
+        } else {
+          res.put(CONTENT, new byte[0]);
+        }
+
+      } else {
+
+        //
+        // Chunk processing
+        //
+
+        if (null != in) {
+
+          int chunkNumber = 0;
```

Comment:
`chunk.number` is 1-indexed. I'm wondering if that's coherent with other indexes in WarpScript.

Reply:
TIMESPLIT for example is 1-indexed, but CHUNK is indexed with the first tick of the partial GTS.

Reply:
OK, that's fine as long as it will be documented ;).